src/meep.hpp

```diff
@@ -1142,6 +1142,11 @@ class dft_chunk {
   // the frequencies to loop_in_chunks
   std::vector<double> omega;
 
+  // decide whether to "persist" after fields class
+  // is deleted (e.g. for adjoint calculations)
+  bool persist = false;
+  bool expand = false;
```

Comment:
Is it necessary to have a separate variable for `expand` since its value is (always?) identical to `persist` (i.e., whenever `persist` is `true` then `expand` must also be `true` and vice versa.). Is there a case where they could have different values?

Reply:
Future applications may require this. I wanted to keep it as general as possible for that reason.

Reply:
This code is already pretty complex, so my suggestion would be to remove one of the duplicated parameters. Having two variables that need to be kept in sync increases the surface area for potentially tricky bugs. If a new parameter is needed in the future, it can be added at that time.

Reply:
Fixed.